packages/syft/src/syft/service/queue/zmq_queue.py

```diff
@@ -476,8 +476,9 @@ def _run(self) -> None:
             if self._stop.is_set():
                 return
 
-            for _, service in self.services.items():
-                self.dispatch(service, None)
+            for service in self.services.keys():
```

Comment:
Should this be

```suggestion
            for service in self.services.values():
```
?

Also let's keep `self.dispatch(service, None)` instead of running it in a thread here. This is a non-trivial change and should be its own PR.

Reply:
https://github.com/OpenMined/PySyft/pull/8901